concent_api/core/views.py

```diff
@@ -1220,6 +1225,632 @@ def store_message_and_message_status(
+                    queue               = PendingResponse.Queue.Receive,
+                    subtask             = subtask,
+                )
+                store_pending_message(
+                    response_type       = PendingResponse.ResponseType.ForceSubtaskResultsSettled,
+                    client_public_key   = subtask.requestor.public_key_bytes,
+                    queue               = PendingResponse.Queue.ReceiveOutOfBand,
+                    subtask             = subtask,
+                )
+            assert subtask.state is not Subtask.SubtaskState.ADDITIONAL_VERIFICATION.name
+
+        logging.log_changes_in_subtask_states(
+            b64encode(client_public_key),
+            clients_subtask_list.count(),
+        )
+    else:
+        logging.log_no_changes_in_subtask_states(
+            b64encode(client_public_key)
+        )
+
+
+def update_subtask_state(
+    subtask,
+    state,
+):
+    logging.log_change_subtask_state_name(
+        subtask.state,
+        state,
+    )
+    subtask.state    = state
+    subtask.next_deadline = None
+    subtask.full_clean()
+    subtask.save()
+
+
+def store_pending_message(
+    response_type       = None,
+    client_public_key   = None,
+    queue               = None,
+    subtask             = None,
+    payment_message     = None,
+):
+    client          = Client.objects.get_or_create_full_clean(client_public_key)
+    receive_queue   = PendingResponse(
+        response_type   = response_type.name,
+        client          = client,
+        queue           = queue.name,
+        subtask         = subtask,
+    )
+    receive_queue.full_clean()
+    receive_queue.save()
+    if payment_message is not None:
+        payment_committed_message = PaymentInfo(
+            payment_ts                  = datetime.datetime.fromtimestamp(payment_message.payment_ts, timezone.utc),
+            task_owner_key_bytes        = decode_key(payment_message.task_owner_key),
+            provider_eth_account_bytes  = payment_message.provider_eth_account,
+            amount_paid                 = payment_message.amount_paid,
+            recipient_type              = payment_message.recipient_type.name,  # pylint: disable=no-member
+            amount_pending              = payment_message.amount_pending,
+            pending_response            = receive_queue
+        )
+        payment_committed_message.full_clean()
+        payment_committed_message.save()
+        subtask_id = None
+    else:
+        subtask_id = subtask.subtask_id
+
+    logging.log_new_pending_response(
+        response_type.name,
+        queue.name,
+        subtask_id,
+        client.public_key,
+    )
+
+
+def handle_messages_from_database(
+    client_public_key:  bytes                   = None,
+    response_type:      PendingResponse.Queue   = None,
+):
+    assert client_public_key    not in ['', None]
+
+    database_message = PendingResponse.objects.filter(
```

Comment:
Could you it `pending_response`? It's not a `StoredMessage` so calling it `database_message` is confusing.